ql/src/java/org/apache/hadoop/hive/ql/exec/tez/DagUtils.java

```diff
@@ -1078,6 +1078,12 @@ public JobConf createConfiguration(HiveConf hiveConf) throws IOException {
 
     JobConf conf = new JobConf(new TezConfiguration(hiveConf));
 
+    conf.set("mapreduce.map.env", "");
+    conf.set("mapreduce.reduce.env", "");
```

Comment:
Add comments about what these lines are doing.

Reply:
The config needs to be set. But there is a bug in HADOOP, where yarn append JAVA_HOME from multiple sources. This is just a workaround to unblock use from using `tst` cluster.

Reply:
I'm not saying it cannot be set. I was asking why it cannot be defined with the hiveConf, which could be xml file based configuration